gluten-data/src/main/java/io/glutenproject/init/NativeBackendInitialized.java

```diff
@@ -19,29 +19,29 @@
 import io.glutenproject.GlutenConfig;
 import io.glutenproject.backendsapi.BackendsApiManager;
 
-import org.apache.spark.sql.internal.SQLConf;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Map;
 
-// Initialize global contexts before calling any native methods from Java side.
-public abstract class JniInitialized {
+// Initialize native backend before calling any native methods from Java side.
+public class NativeBackendInitialized {
 
-  private static final Logger LOG = LoggerFactory.getLogger(JniInitialized.class);
+  private static final Logger LOG = LoggerFactory.getLogger(NativeBackendInitialized.class);
 
-  static {
+  public static void initializeBackend(scala.collection.Map<String, String> conf) {
     try {
       String prefix = BackendsApiManager.getSettings().getBackendConfigPrefix();
-      Map<String, String> nativeConfMap =
-          GlutenConfig.getNativeBackendConf(prefix, SQLConf.get().getAllConfs());
```

Comment:
I found `SQLConf.get().getAllConfig` return empty map in some test cases.

Reply:
it seems break the config propagation after moving the initialization to the beginning. For example, if people first call `spark.sql("set a=b")` then run query, then the native side can not aware of the config.

Reply:
The initialization here is for executor level and only pass need static config to VeloxBackend once, other dynamic config will be passed to native WholeStageIter in nativeCreateKernelWithIterator.

If `set a = b` is needed when initialize VeloxBackend, it should be added before intialize Spark session.

Reply:
In addition, I suppose `set a = b` would cause an error when `a` is a "static conf key" in vanilla Spark. However in Gluten we didn't yet have a way to tell whether a Gluten config key is static or not. Probably we should somehow implement that.

Reply:
make sense to me, thank you